### DISCUSSION TITLE
Is it recommended to have a single scheduled job for all assets running on a given cadence?

### DISCUSSION THREAD

**Discussion by rob-apella** [dagster-io/dagster#17833] [Q&A] [answered]
I have a large collection of assets that currently run every day on a daily schedule. These assets are organized into groups, and there are cross-group dependencies between assets. 

As we onboard new assets that also need to run daily, is it recommended to just add them to a single master daily schedule, or should we think about creating schedules per asset grouping? Would cross-group dependencies still work in this case?

**Comment by jamiedemaria** [comment, marked as answer]
If the new assets don't have any shared dependencies with the assets in the existing schedule, you may find some benefits in introducing new schedules (and jobs) for the new assets. Some reasons for this are:
* if some group of assets need to be not run daily (for example if a bug is found and you want to wait to materialize the assets until it is fixed) having multiple schedules will allow you to just turn off the affected assets and allow the others to continue materializing. 
* Multiple smaller jobs with a single purpose can cause less confusion if you need to do a manual run of one of the jobs

However, this is mostly a matter of preference and if you think having a single job that's run from one daily schedule will be easier to maintain or has other pros that outway the pros of multiple schedules, then you can absolutely do that

**Reply by rob-apella** [reply]
@jamiedemaria coming back to this after some time. we are encountering this dependency issue when going the route of multiple smaller jobs:

https://github.com/dagster-io/dagster/discussions/28023

do you have any advice here on how we can make smaller jobs work with asset dependencies between jobs?
